protographic/src/operations/field-numbering.ts

```diff
@@ -0,0 +1,204 @@
+import { ProtoLockManager } from '../proto-lock.js';
+
+/**
+ * Field numbering manager for Protocol Buffer messages
+ *
+ * This module handles the assignment and tracking of field numbers
+ * across multiple proto messages to ensure uniqueness within each message.
+ * Integrates with ProtoLockManager for field number stability across compilations.
+ */
+
+/**
+ * Manages field number assignment for a collection of messages
+ */
+export interface FieldNumberManager {
+  /**
+   * Gets the next available field number for a given message
+   * @param messageName - The name of the message
+   * @returns The next available field number
+   */
+  getNextFieldNumber(messageName: string): number;
+
+  /**
+   * Assigns a specific field number to a field in a message
+   * @param messageName - The name of the message
+   * @param fieldName - The name of the field
+   * @param fieldNumber - The field number to assign
+   */
+  assignFieldNumber(messageName: string, fieldName: string, fieldNumber: number): void;
+
+  /**
+   * Gets the field number for a specific field if it exists
+   * @param messageName - The name of the message
+   * @param fieldName - The name of the field
+   * @returns The field number or undefined if not assigned
+   */
+  getFieldNumber(messageName: string, fieldName: string): number | undefined;
+
+  /**
+   * Resets field numbering for a specific message
+   * @param messageName - The name of the message to reset
+   */
+  resetMessage(messageName: string): void;
+
+  /**
+   * Resets all field numbering
+   */
+  resetAll(): void;
+
+  /**
+   * Gets all field mappings for a message
+   * @param messageName - The name of the message
+   * @returns Record of field names to field numbers
+   */
+  getMessageFields(messageName: string): Record<string, number>;
+
+  /**
+   * Reconciles field order for a message using lock data
+   * @param messageName - The name of the message
+   * @param fieldNames - The field names to reconcile
+   * @returns Ordered array of field names
+   */
+  reconcileFieldOrder(messageName: string, fieldNames: string[]): string[];
+
+  /**
+   * Gets the lock manager if available
+   */
+  getLockManager?(): ProtoLockManager | undefined;
+}
+
+/**
+ * Creates a new field number manager instance
+ *
+ * @param lockManager - Optional ProtoLockManager for field number stability
+ * @returns A new field number manager
+ */
+export function createFieldNumberManager(lockManager?: ProtoLockManager): FieldNumberManager {
```

Comment:
How will this integrate with the current connect approach? Will you create a separate lock file or will you touch an existing one?

Reply:
My thoughts were that they are two separate APIs / servers with separate purposes. So therefore they should have separate lock files.

SDL to Proto - Conenct API converts GraphQL SDL (federation schema) to proto.
Operations to proto / Persisted Operations converts individual named operations to proto - the proto depends on each operation's shape.

I didn't think too much further - but my approach was:

One lock file per service / proto file.
Each proto generation maintains its own lock
The two systems are independent - SDL to proto lock shouldn't interfere with operations to proto lock - as they are different APIs.

Both use the same ProtoLockManager but maintain separate lock data.